core/domain/exp_jobs_one_off.py

```diff
@@ -301,6 +325,106 @@ def reduce(key, values):
         yield (key, values)
 
 
+class ExplorationMathTagValidationOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that checks the html content of an exploration and validates all the
+    Math tags in the HTML.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        exploration = exp_fetchers.get_exploration_from_model(item)
+        exploration_status = (
+            rights_manager.get_exploration_rights(
+                item.id).status)
+        for state_name, state in exploration.states.items():
+            html_string = ''.join(state.get_all_html_content_strings())
+            error_list = (
+                html_validation_service.validate_math_tags_in_html(html_string))
+            if len(error_list) > 0:
+                key = (
+                    'exp_id: %s, exp_status: %s failed validation.' % (
+                        item.id, exploration_status))
+                value_dict = {
+                    'state_name': state_name,
+                    'error_list': error_list,
+                    'no_of_invalid_tags': len(error_list)
+                }
+                yield (key, value_dict)
+
+    @staticmethod
+    def reduce(key, values):
+        yield (key, values)
+
+
+class ExplorationMockMathMigrationOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that migrates all the math tags in the exploration to the new schema
+    but does not save the migrated exploration. This job is used to verify
+    that the actual migration will be possible for all the explorations.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        exploration = exp_fetchers.get_exploration_from_model(item)
+        exploration_status = (
+            rights_manager.get_exploration_rights(
+                item.id).status)
+        for state_name, state in exploration.states.items():
+            try:
+                converted_state_dict = (
+                    state_domain.State.convert_html_fields_in_state(
+                        state.to_dict(),
+                        html_validation_service.
+                        add_math_content_to_math_rte_components))
+                with mock_swap(html_cleaner, 'clean', lambda html: html):
```

Comment:
If we aren't going to run this function then can we just do this:

1. Get all html strings from state
2. Run add_math_content_to_math_rte_components on the html strings?

[We won't need the mock_swap for this as well.]

Reply:
I just wanted to check whether the convert_html_fields_in_state works for all the data in the backup server, that's why I tried to do the complete migration. Also If this job passes, we can be almost sure the migration will pass.
Anyway let's see what @seanlip, has to say.

Reply:
Basically in the current state the job ExplorationMathTagValidationOneOffJob does the following:
1. convert all the math-tags in the HMTL to the new schema.
2. validate the state after conversion.
3. extract all the HTML from the converted state
4. Validate all the math-tags in the extracted HTML to have the new schema.

Reply:
Please get rid of mock_swap -- that isn't the way to do it.

What @DubeySandeep is what I thought you should do. We know the rest of the migration works, you just need to fully verify the bits that depend on the new conversion function you are writing.

If you want to verify the html after the fact then swap out the relevant part of html_cleaner using the existing self.swap() infrastructure. Don't invent your own mock_swap.

Reply:
done